### ISSUE TITLE
unit-http Transfer-Encoding being duplicated

### ISSUE THREAD

**Issue by dbit-xia** [nginx/unit#1088]
When nodejs-app already includes `Transfer-Encoding` in the response headers, 
Unit will add an additional one, causing abnormal forwarding in Traefik. 

The issue can be found at: https://github.com/traefik/traefik/issues/8623.

![image](https://github.com/nginx/unit/assets/19259623/3ce10d58-2a40-4120-ba7a-90ca67eea00d)

**Comment by tippexs** [comment]
Hi @dbit-xia thanks for reaching out and sorry for the long delay in my answer! Currently the behavior is not documented elsewere as in the code when a `Transfer-Encoding` header will be added or not by Unit.

Can you make a request to the NodeJS Application outside of Unit and check if it has a `Content-Length`. As you are saying that Node or better Express sets the `Transfer-Encoding: chunked` header I guess there will be no `Content-Length`. But would be worth a test.

[RFC7230](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.2) just says that the Server might combine headers with the same values. So thats what we do with NGINX for example.

Traeffic is saying this is an issue from the backend and thats why they are closing it. Good. The RFC is saying multiple headers with the same value are allowed and should be combined if that is not impacting the overall request processing. 

Long Story short. Unit sets the `Transfer-Encoding` header because there is not `Content-Length` Header and in the code we are currently not checking if there is already an `Transfer-Enconding` Header from the application in the shared context. This is something we have to investigate if that would be a possible solution.

**Comment by callahad** [comment]
Assigning to @tippexs to investigate and characterize Unit's behavior; this does seem like a bug on our end, so marking a T-Defect

**Comment by willnode** [comment]
This problem is more likely bigger than it seems. I conduct a test and found that with express, using `res.end()` instead of `res.send()` will make the response chunked thus the double header returned. If my unit app is also behind NGINX then it simply returns nothing.

I made a workaround for myself by creating [a proxy](https://github.com/domcloud/proxy-unit/commit/0c64032).

**Comment by koushyk** [comment]
any updates I have duplicated headers with laravel app

**Comment by ac000** [comment]
Not really no.

Do we know where this chunked header is coming from? (Not the one Unit adds, the other one).
